src/services/api.ts

```diff
@@ -77,11 +105,28 @@ const appBaseQuery: BaseQueryFn<
     return { data: result.data, meta };
   } catch (error) {
     if (isAxiosError(error)) {
+      if (error.response?.status === 401) {
+        console.debug("Invalid token, resetting extension auth");
+        // The token is bad, clear it out
+        await clearExtensionAuth();
+      }
+
+      // Axios offers its own serialization method, but it doesn't include the response.
+      // By deleting toJSON, the serialize-error library will use its default serialization
+      delete error.toJSON;
+
       return {
```

Comment:
Since we're returning the whole error, maybe we don't need to _also_ pass `status` and `data` separately? Maybe just:

```suggestion
      return serializeError(error);
```

Reply:
IIRC, RTK might expect status and data? https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#fetchbasequery-defaults

Although re-reading the documentation I think we can pass our own error. We'd just need to make sure we're handling it properly across all queries

Reply:
Created a ticket here: https://github.com/pixiebrix/pixiebrix-extension/issues/3126